File: router/relation.go
package router

import (
	"github.com/gin-gonic/gin"
	"github.com/intyouss/Traceability/api"
)

func InitRelationRoutes() {
	RegisterRoute(func(dfGroup *gin.RouterGroup, auGroup *gin.RouterGroup) {
		relationApi := api.NewRelationApi()
		//relationDefaultGroup := dfGroup.Group("relation")
		//{
		//
		//}
		relationAuthGroup := auGroup.Group("relation")
		{
			// 关注/取消关注
			relationAuthGroup.POST("/action", relationApi.RelationAction)
			// 关注列表
			relationAuthGroup.GET("/focus/list", relationApi.GetFocusList)
			// 粉丝列表
			relationAuthGroup.GET("/fans/list", relationApi.GetFansList)
		}
	})
}

File: web/src/api/relation.js
import {
  authAPI,
} from '~/axios';

/**
 * 关注/取关
 * @param {Number} userId
 * @param {Number} actionType
 * @return {Promise<AxiosResponse<any>>}
 */
export function relationAction(userId, actionType) {
  return authAPI.post('/relation/action', {
    'user_id': userId,
    'action_type': actionType,
  });
}

/**
 * 获取关注列表
 * @param {Number} userId
 * @return {Promise<AxiosResponse<any>>}
 */
export function getFocusList(userId) {
  return authAPI.get('/relation/focus/list', {params: {
    'user_id': userId,
  }});
}

/**
 * 获取粉丝列表
 * @param {Number} userId
 * @return {Promise<AxiosResponse<any>>}
 */
export function getFansList(userId) {
  return authAPI.get('/relation/fans/list', {params: {
    'user_id': userId,
  }});
}
